js/adapt-assessmentArticleModel.js

```diff
@@ -623,9 +624,10 @@ define([
                 this.set({'_attemptsLeft':this.get('_attempts')});
                 this.set({'_attemptsSpent':0});
             }
-
-            //stop resetting if no attempts left
-            if (!this._isAttemptsLeft() && !force) {
+            
+            var isAllowResetIfPassed = this.get('_assessment')._allowResetIfPassed;
```

Comment:
This would read much better if left as `allowResetIfPassed` or `shouldAllowResetIfPassed`.